### ISSUE TITLE
Suspicious comment (//) adjacent to doc comment (///)

### ISSUE THREAD

**Issue by alice-i-cecile** [rust-lang/rust-clippy#9213]
### What it does

If a `//` comment is found adjacent to a doc comment, warn, as it may have been intended to be part of the doc comment.

Some cases that definitely should be warned about:
- a `//` comment above a `///` doc comment empty line
- a `//` comment (either single or multi-line) surrounded by doc comment lines

Some edge cases:
- a `//` comment before a doc comment
- a `// comment after a doc comment

Some cases that should definitely be excluded:
- comments beginning with an all-caps word, like `TODO`, `FIXME`, `PERF`
- comments defined by `/* */`, as that is much harder to typo

### Lint Name

suspicious-comment-beside-doc-comment

### Advantage

Ordinary comments do not appear in generated documentation.

### Drawbacks

1. It becomes harder to comment on the documentation itself. This is a pretty rare pattern; usually I'd use issue trackers for that sort of thing.
2. The exact rule should be tuned to make sure that doc comments on a struct / function can exist beside ordinary comments without triggering false positives.

### Example

```rust
    // Consumes the [`OwningPtr`] to obtain ownership of the underlying data of type `T`.
    ///
    /// # Safety
    /// Must point to a valid `T`.
```

Was likely intended to be

```rust
    /// Consumes the [`OwningPtr`] to obtain ownership of the underlying data of type `T`.
    ///
    /// # Safety
    /// Must point to a valid `T`.
```

**Comment by alice-i-cecile** [comment]
Another example of this today: https://github.com/bevyengine/bevy/pull/14894#discussion_r1770568522
